strawberry/ext/mypy_plugin.py

```diff
@@ -262,33 +274,42 @@ def is_dataclasses_field_or_strawberry_field(expr: Expression) -> bool:
     return False
 
 
-def _collect_field_args(expr: Expression) -> Tuple[bool, Dict[str, Expression]]:
+def _collect_field_args(
+    expr: Expression, ctx: ClassDefContext
+) -> Tuple[bool, Dict[str, Expression]]:
     """Returns a tuple where the first value represents whether or not
     the expression is a call to dataclass.field and the second is a
     dictionary of the keyword arguments that field() was called with.
     """
-
-    if is_dataclasses_field_or_strawberry_field(expr):
-        expr = cast(CallExpr, expr)
-
+    if (
+        isinstance(expr, CallExpr)
+        and isinstance(expr.callee, RefExpr)
+        and expr.callee.fullname == "dataclasses.field"
+    ):
```

Comment:
I think we should keep using `is_dataclasses_field_or_strawberry_field` otherwise it won't work for `strawberry.field` :)

Reply:
I undid all the changes to the function, seems like they were just adding an error message which seems irrelevant.